src/transaction/transaction.ts

```diff
@@ -310,279 +375,377 @@ async function getUnnamedAppCallResourcesAccessed(atc: algosdk.AtomicTransaction
  *
  */
 export async function populateAppCallResources(atc: algosdk.AtomicTransactionComposer, algod: algosdk.Algodv2) {
-  const unnamedResourcesAccessed = await getUnnamedAppCallResourcesAccessed(atc, algod)
+  return await alterGroupBasedOnSendParams(atc, algod, { populateAppCallResources: true })
+}
+
+/**
+ * Take an existing Atomic Transaction Composer and return a new one with alterations
+ * based on the supplied sendParams to ensure the transaction group is valid for sending.
+ *
+ * @param algod The algod client to use for the simulation
+ * @param atc The ATC containing the txn group
+ * @param sendParams The send params for the transaction group
+ * @param executionContext Additional execution context used to determine how best to alter transactions in the group
+ * @returns A new ATC with the alterations applied
+ *
+ * @privateRemarks
+ * Parts of this function will eventually be implemented in algod. Namely:
+ * - Simulate will return information on how to populate reference arrays, see https://github.com/algorand/go-algorand/pull/6015
+ */
+async function alterGroupBasedOnSendParams(
```

Comment:
Should we expose this like we do with `populateAppCallResources`? It's possible that devs will want to calculate fees, populate etc without sending immediately.

Also open to suggestions for a better name.

If we did expose this, we could potentially deprecate `populateAppCallResources`, as it's quite likely devs will want to use these features together.